packages/popup/src/ux-popup.ts

```diff
@@ -0,0 +1,157 @@
+import { customElement, inject, bindable, PLATFORM, TaskQueue, useView } from 'aurelia-framework';
+import { UxComponent, StyleEngine, normalizeBooleanAttribute } from '@aurelia-ux/core';
+import { UxPopupTheme } from './ux-popup-theme';
+
+const windowEvents = ['click', 'wheel', 'scroll', 'resize'];
+
+interface IAnchor {
+  left: string | undefined;
+  right: string | undefined;
+  top: string | undefined;
+  bottom: string | undefined;
+  maxHeight: number | undefined;
+  maxWidth: number | undefined;
+}
+
+@inject(Element, StyleEngine, TaskQueue)
+@customElement('ux-popup')
+@useView(PLATFORM.moduleName('./ux-popup.html'))
+export class UxPopup implements UxComponent, EventListenerObject {
+  constructor(private element: HTMLElement, private styleEngine: StyleEngine, private taskQueue: TaskQueue) { }
+
+  public anchor: IAnchor | null;
+  public isWrapperOpen: boolean = false;
+  public isOpen: boolean = false;
+  public popupContent: HTMLDivElement;
+  public isMeasured: boolean = false;
+
+  @bindable
+  trigger: HTMLElement;
```

Comment:
I think it's also reasonable to support the usage:
```html
<ux-popup trigger="#someid">
```
For that, we may need to tweak this trigger abit

Reply:
done